datafusion/physical-expr/src/aggregate/approx_percentile_cont.rs

```diff
@@ -284,7 +284,9 @@ impl ApproxPercentileAccumulator {
     }
 
     pub(crate) fn merge_digests(&mut self, digests: &[TDigest]) {
-        self.digest = TDigest::merge_digests(digests);
+        let mut input_digests = digests.to_vec();
+        input_digests.push(self.digest.clone());
```

Comment:
I think it is possible to avoid these clones -- here is a proposal that targets this PR https://github.com/Ted-Jiang/arrow-datafusion/pull/118 to do so